airflow/creds/__init__.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Creds framework provides means of getting connection objects from various sources e.g. the following:
+* Environment variables
+* Metatsore database
+* AWS SSM Parameter store
+"""
+
+from abc import ABC, abstractmethod
+from typing import List
+
+from airflow import AirflowException, conf
+from airflow.models import Connection
+from airflow.utils.module_loading import import_string
+
+CONN_ENV_PREFIX = "AIRFLOW_CONN_"
+
+
+class BaseCredsBackend(ABC):
+    """
+    Abstract base class to provide connection objects given a conn_id
```

Comment:
I think the class- name and description here highlight the difference between the naming of the feature and it's intent, as the description talks about connection objects and the class name about a creds backend.

Reply:
That is a good point actually !

Reply:
i spoke to the thinking [in this comment](https://github.com/apache/airflow/pull/6376#discussion_r389202690).  i am not attached to the name.  does group feel `BaseConnectionBackend` is the right name?

alternatively can update description to be more compatible.
e.g. 
```
Abstract base class to retrieve creds given a conn_id and construct a Connection object
```

Reply:
I like Credentials name in this context and with the updated description

I think it's a nice distinction - Connections are stored in Airflow DB or variable and Credentials are a way to retrieve the connections. But I can be convinced otherwise.

Another option is to simply make it "SecretsBackend". I think that would be more generic and we could even provide a more generic interface to it, allowing to not only retrieve connections. I will discuss it at the devlist.